### PR TITLE
is-map, is-set: remove no-longer-needed types

### PR DESCRIPTION
**Pull request by ljharb**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).

If removing a declaration:

- [x] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [x] Delete the package's directory.
- [x] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by ljharb** [comment]
it fails with `Error: The specified version 2.0.3 of is-map must be newer than the version it is supposed to replace, 2.0.3 of @types/is-map.`, but i'm not sure what to tell you - v2.0.3 adds inline types.

**Comment by jakebailey** [comment]
This part of the infra is weird; it's talking about https://www.npmjs.com/package/@types/is-map itself, so needs to be v2.0.4. I agree this needs some rethinking but I personally don't have the historical context for why this needs to be this way 😄

**Comment by ljharb** [comment]
oh lol, that is pretty silly :-) updated.

**Comment by jakebailey** [comment]
Okay, what the heck

**Comment by ljharb** [comment]
> Error: The specified version 2.0.4 of is-map is not on npm.

¯\\\_(ツ)_/¯

**Comment by jakebailey** [comment]
This is some weird edge case or something. The types version and the real version are the same, v2.0.3, and the tooling wants >2.0.3, but it can't have that because it's not on npm. Really I don't understand why we're checking anything in relation to the existing types package at all. @sandersn @andrewbranch any ideas?

**Comment by ljharb** [comment]
Alternatively, if the versions of both matter, then there could be an optional additional field in the json that contains the version of the DT package, if different (or the real one, doesn't matter which is which)

**Comment by jakebailey** [comment]
I don't think they matter; IIRC all that happens is that we publish a package which `*` deps on the name given, and then deprecate it with a custom message saying "not needed as of vX.Y.Z" and that's it.

**Comment by LeoDog896** [comment]
glad to see that `inspect-js` is getting its small types! no clue as to why that versioning error is happening, but once the tests are fixed, I'll approve this 👍

File: notNeededPackages.json
Comment (sandersn) [inline notNeededPackages.json:2534]: The error is a mistake in our infrastructure, but we've never had a deprecation that matched `major.minor` before, and also happened to have more patch publishes to the types package than the source package did. We could split the greater-than version from the please-use version but the deprecation checks+message is a borderline misfeature [1]. A fix is not worth it in the general case unless it happens a few more times, and it's definitely not worth it in this case, where the types packages usage is a rounding error compared to usage of the source packages (~50k/month vs ~70M/month).

I'd recommend reverting all changes to notNeededPackages.json and doing nothing but deleting is-map/is-set files. This will signal to DT maintainers not to publish types from here anymore, but leave the already-published packages unaffected.

[1] tsc checks source package locations first, so a duplicate install of package `x` and `@types/x` ignores `@types/x`. Plus, publishing a deprecated package as `latest` can break some dependents of it.

**Review by sandersn — CHANGES_REQUESTED** [review]

File: notNeededPackages.json
Comment (ljharb) [reply notNeededPackages.json]: oops, thanks.

it was my mistake, but `pnpm run not-needed` was happy to add these anyways :-)

**Review by sheetalkamat — APPROVED** [review]

**Comment by ljharb** [comment]
Ready to merge

**Pull request merged**
